lib/authenticate.js

```diff
@@ -51,6 +51,11 @@ var authenticate = function(client, username, password, options, callback) {
       if (err) return handleCallback(callback, err, false);
       _callback(null, true);
     });
+  } else if (authMechanism === 'SCRAM-SHA-256') {
+    client.topology.auth('scram-sha-256', authdb, username, password, function(err) {
```

Comment:
it looks like every one of these methods does exactly the same thing, only changing the first string parameter. Can we combine them? It would certainly make refactoring easier in the future.

Reply:
My plan was to do that as a refactor once this went in :).